cypress/components/dialog/dialog.cy.tsx

```diff
@@ -122,21 +122,15 @@ context("Testing Dialog component", () => {
     });
 
     it("should call the cancel action after the CloseIcon is clicked", () => {
-      const callback = cy.stub();
-
+      const callback: DialogProps["onCancel"] = cy.stub().as("onCancel");
       CypressMountWithProviders(
         <stories.DialogComponent onCancel={callback} />
       );
-
-      closeIconButton()
-        .click()
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      closeIconButton().click();
+      cy.get("@onCancel").should("have.been.calledOnce");
     });
 
-    it.each([["top"], ["topRight"], ["right"]])(
+    it.each(["top", "topRight", "right"] as ("top" | "topRight" | "right")[])(
```

Comment:
```suggestion
    it.each(["top", "topRight", "right"] as Cypress.PositionType[])(
```

**suggestion**: This seems to work and avoids the repetition of the values. This is a type defined within Cypress, for the arguments that can be given to `.click`.